File: drills/python/harnesses/nc27_eval_rpn.py
try:
    (evalRPN)
except NameError:
    raise Exception("__signature_mismatch__")

__results__ = []
def __case__(label, expected, actual):
    __results__.append([label, repr(expected), repr(actual)])

__case__("evalRPN(['2', '1', '+', '3', '*'])", 9, evalRPN(["2", "1", "+", "3", "*"]))
__case__("evalRPN(['4', '13', '5', '/', '+'])", 6, evalRPN(["4", "13", "5", "/", "+"]))
__case__("evalRPN(['-3', '2', '/'])", -1, evalRPN(["-3", "2", "/"]))
__case__("evalRPN(['5'])", 5, evalRPN(["5"]))
__case__("evalRPN(the long one)", 22, evalRPN(["10", "6", "9", "3", "+", "-11", "*", "/", "*", "17", "+", "5", "+"]))

File: drills/go/harnesses/nc27_eval_rpn.go
package main

func main() {
	run(func() []testCase {
		return []testCase{
			tc("evalRPN(['2','1','+','3','*'])", 9, evalRPN([]string{"2", "1", "+", "3", "*"})),
			tc("evalRPN(['4','13','5','/','+'])", 6, evalRPN([]string{"4", "13", "5", "/", "+"})),
			tc("evalRPN(['10','6','9','3','+','-11','*','/','*','17','+','5','+'])", 22, evalRPN([]string{"10", "6", "9", "3", "+", "-11", "*", "/", "*", "17", "+", "5", "+"})),
			tc("evalRPN(['7','-3','/'])", -2, evalRPN([]string{"7", "-3", "/"})),
		}
	})
}
